### ISSUE TITLE
boxedwine会添加对于64位的binary的支持吗(would boxedwine have the ability to lanuch 64bit binary)

### ISSUE THREAD

**Issue by flyingfish120** [danoon2/Boxedwine#55]
现阶段boxedwine仅支持wine32，将来会添加wine64的支持吗(
Currently boxedwine only supports wine32, will wine64 support be added in the future?)

**Comment by danoon2** [comment]
No, I currently have no plans for wine64.  I think that might be so different as to be a different project.  The CPU emulation would be a lot different and even the syscall interfaces are a little different.

**Comment by duckfromdiscord** [comment]
Out of curiosity, what would be different for wine64 to work? Also what is x64CPU for?

**Comment by danoon2** [comment]
x64CPU is used to run 32-bit emulated code on a 64-bit x64 cpu (Armv8bt is the 64-bit Arm version for things like Raspberry Pi and Mac M1).  It's a binary translator so some instruction like "add eax, ecx" would stay the same, some instruction, like ones that use memory need a bit of work.  But overall it's pretty fast.

Wine 64 would require new cpu emulation and a new set of syscall mappings.  My goal is to run late 90's and maybe early 2000's games so I wouldn't even consider 64-bit emulation at this time.  If a miracle happens and I get everything working very well for 32-bit then who knows, but I think I would be pretty old by then :)

**Comment by danoon2** [comment]
And supporting a larger memory space would be a challenge, I already use way too much memory just to run a simple app

**Comment by Spidy123222** [comment]
> x64CPU is used to run 32-bit emulated code on a 64-bit x64 cpu (Armv8bt is the 64-bit Arm version for things like Raspberry Pi and Mac M1).  It's a binary translator so some instruction like "add eax, ecx" would stay the same, some instruction, like ones that use memory need a bit of work.  But overall it's pretty fast.
> 
> 
> 
> Wine 64 would require new cpu emulation and a new set of syscall mappings.  My goal is to run late 90's and maybe early 2000's games so I wouldn't even consider 64-bit emulation at this time.  If a miracle happens and I get everything working very well for 32-bit then who knows, but I think I would be pretty old by then :)

Would using fex-emu be similar system calls as whatever boxedwine uses. They do need host to be arm64 but is Linux focused.
